lib/dice/src/cert.rs

```diff
@@ -0,0 +1,346 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use crate::{
+    alias_cert_tmpl, deviceid_cert_tmpl, SerialNumber, NOTBEFORE_LENGTH,
+};
+use hubpack::SerializedSize;
+use salty::constants::{
+    PUBLICKEY_SERIALIZED_LENGTH, SIGNATURE_SERIALIZED_LENGTH,
+};
+use salty::signature::{Keypair, PublicKey};
+use serde::{Deserialize, Serialize};
+use serde_big_array::BigArray;
+use unwrap_lite::UnwrapLite;
+
+#[derive(Debug)]
+pub enum CertError {
+    BadSig,
+    NoPubKey,
+    NoSig,
+    NoSignData,
+    TooSmall,
+    NotFound,
+    NoCn,
+}
+
+pub trait Cert {
+    fn as_bytes(&self) -> &[u8];
+    fn as_mut_bytes(&mut self) -> &mut [u8];
+
+    const SERIAL_NUMBER_START: usize;
+    const SERIAL_NUMBER_END: usize;
+
+    fn get_serial_number(&self) -> u8 {
+        u8::from_be_bytes(
+            self.as_bytes()[Self::SERIAL_NUMBER_START..Self::SERIAL_NUMBER_END]
+                .try_into()
+                .unwrap(),
+        )
+    }
+
+    fn set_serial_number(mut self, sn: u8) -> Self
+    where
+        Self: Sized,
+    {
+        self.as_mut_bytes()[Self::SERIAL_NUMBER_START..Self::SERIAL_NUMBER_END]
+            .copy_from_slice(&sn.to_be_bytes());
+
+        self
+    }
+
+    const ISSUER_SN_START: usize;
+    const ISSUER_SN_END: usize;
+
+    fn get_issuer_sn(&self) -> SerialNumber {
+        SerialNumber::from_bytes(
+            &self.as_bytes()[Self::ISSUER_SN_START..Self::ISSUER_SN_END]
+                .try_into()
+                .unwrap_lite(),
```

Comment:
Why are we using `unwrap_lite` here and `unwrap` at line 39?

Reply:
I think I only found `unwrap_lite` as I was doing the implementation here. I thought I had replaced all of the `unwrap`s with `unwrap_lite`s but looks like I missed at least one. I'll take a pass over the code to make sure there aren't others.